zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/DocumentenApiClient.kt

```diff
@@ -72,6 +77,75 @@ class DocumentenApiClient(
         return result
     }
 
+     fun storeDocumentInParts(
+        authentication: DocumentenApiAuthentication,
+        baseUrl: URI,
+        request: BestandsdelenRequest,
+        createDocumentResult: CreateDocumentResult,
+        bestandsnaam: String?
+    ): BestandsdelenResult {
+         requireNotNull(bestandsnaam) { "Bestandsnaam must be set otherwise uploading in bestanddelen will fail" }
+
+         // Inside the CreateDocumentResult there is an array of bestandsdelen.
+         // Each bestandsdeel needs to be sent separately
+         // So the documenten api determines the amount of chunks, not this application.
+         val fileBytes = request.inhoud.readAllBytes()
```

Comment:
This can be problematic. This loads the entire file into memory, so if the file is too large (or too many files are uploaded at the same time), we'll get an OutOfMemoryError.

Reply:
What would be the solution to this? The documentten api expects a binary not a stream

Reply:
`request.inhoud` is an `InputStream`. Instead of reading all bytes you can just read the ones you need to send.

Inside of the for-loop you could use something like this:

```
val chunk = ByteArray(bestandsdeel.omvang)
request.inhoud.read(chunk)
```

After that, chunk will contain the data that is read from the InputStream.

This does assume that that the sum of all `bestandsdeel.omgvang` add up to the total size of bytes that can be read (same as the code does now). The `InputStream.read(byte[])` method returns the number of bytes read, so it could be wise to assert that the return value is equal to the `chunk.size`. Also at the end you can use the `.read()` method of the InputStream [to see if it returns `-1`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/InputStream.html#read()) to ensure that the end of the stream has been reached.